sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/impl/utils/CalciteUtilsTest.java

```diff
@@ -59,68 +64,121 @@ public void setUp() {
   public void testToCalciteRowType() {
     final Schema schema =
         Schema.builder()
-            .addField("f1", Schema.FieldType.BYTE)
-            .addField("f2", Schema.FieldType.INT16)
-            .addField("f3", Schema.FieldType.INT32)
-            .addField("f4", Schema.FieldType.INT64)
-            .addField("f5", Schema.FieldType.FLOAT)
-            .addField("f6", Schema.FieldType.DOUBLE)
-            .addField("f7", Schema.FieldType.DECIMAL)
-            .addField("f8", Schema.FieldType.BOOLEAN)
-            .addField("f9", Schema.FieldType.BYTES)
-            .addField("f10", Schema.FieldType.STRING)
+            .addField("byte", Schema.FieldType.BYTE)
+            .addField("short", Schema.FieldType.INT16)
+            .addField("int", Schema.FieldType.INT32)
+            .addField("long", Schema.FieldType.INT64)
+            .addField("float", Schema.FieldType.FLOAT)
+            .addField("double", Schema.FieldType.DOUBLE)
+            .addField("decimal", Schema.FieldType.DECIMAL)
+            .addField("boolean", Schema.FieldType.BOOLEAN)
+            .addField("byteArray", Schema.FieldType.BYTES)
+            .addField("string", Schema.FieldType.STRING)
+            .addField("char", CalciteUtils.CHAR)
+            .addField("date", CalciteUtils.DATE)
+            .addField("time", CalciteUtils.TIME)
+            .addField("timestamp", CalciteUtils.TIMESTAMP)
+            .addField("timeWithLocalTZ", CalciteUtils.TIME_WITH_LOCAL_TZ)
+            .addField("timestampWithLocalTZ", CalciteUtils.TIMESTAMP_WITH_LOCAL_TZ)
+            .addField("fixedBytes", Schema.FieldType.logicalType(FixedBytes.of(10)))
+            .addField("variableBytes", Schema.FieldType.logicalType(VariableLengthBytes.of(100)))
+            .addField("fixedString", Schema.FieldType.logicalType(FixedLengthString.of(10)))
+            .addField("variableString", Schema.FieldType.logicalType(VariableLengthString.of(100)))
+            .addField("customDecimal", Schema.FieldType.logicalType(LogicalDecimal.of(10, 5)))
```

Comment:
Could you add tests that use some or all of these new types in a SQL query? Right now I think this is the only test that exercises the new code in CalciteUtils, and it's not actually verifying the types will work in a SQL statement.